### ISSUE TITLE
MinGW not usable because there is Gits sh.exe in the PATH

### ISSUE THREAD

**Issue by Flamefire** [actions/virtual-environments#111]
**Describe the bug**
Runing  `cmake` with ` -G "MinGW Makefiles"` fails with the following error:
```
CMake Error at C:/Program Files/CMake/share/cmake-3.15/Modules/CMakeMinGWFindMake.cmake:12 (message):
  sh.exe was found in your PATH, here:

  C:/Program Files/Git/bin/sh.exe

  For MinGW make to work correctly sh.exe must NOT be in your path.

  Run cmake from a shell that does not have sh.exe in your PATH.

  If you want to use a UNIX shell, then use MSYS Makefiles.
```

**Virtual environments affected**
- [ ] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [x] Windows Server 2016 R2
- [x] Windows Server 2019

**Expected behavior**
MinGW usable out of the box

See also: https://help.appveyor.com/discussions/problems/3193-cmake-building-for-mingw-issue-with-git-shexe

**Comment by zauonlok** [comment]
Same issue here. I managed to get MinGW working by adding these two lines:
```
set PATH=%PATH:C:\Program Files\Git\bin;=%
set PATH=%PATH:C:\Program Files\Git\usr\bin;=%
```

See also:
https://github.com/zauonlok/renderer/blob/6c274ed54f758c54b1ee4f9e3715e9991601be4d/.github/workflows/cmake-ci.yml#L22-L23

**Comment by Flamefire** [comment]
Note that this can be work-arounded by using the current CMake version or passing `-DCMAKE_SH=SH-NOT-FOUND` to cmake

**Comment by zauonlok** [comment]
Yes, this issue is fixed on the master branch of CMake: https://github.com/Kitware/CMake/commit/82ddcf0db1d220564145122c3cce25d25ee0e254

**Comment by zauonlok** [comment]
I have updated my workflow file. It works pretty well. https://github.com/zauonlok/renderer/blob/a31f31511ecaaa9edc44f0a82757f91d4a8d87c0/.github/workflows/cmake-ci.yml#L24

**Comment by alepauly** [comment]
This was fixed in #211 which should be deployed next week.
